packages/upload-client/src/types.ts

```diff
@@ -87,10 +83,13 @@ import {
 type Override<T, R> = Omit<T, keyof R> & R
 
 type FetchOptions = Override<
-  IpfsUtilsFetchOptions,
+  RequestInit,
   {
-    // `fetch` is a browser API and browsers don't have `Readable`
-    body: Exclude<IpfsUtilsFetchOptions['body'], import('node:stream').Readable>
+    /**
+     * Can be passed to track upload progress.
+     * Note that if this option in passed underlying request will be performed using `XMLHttpRequest` and response will not be streamed.
```

Comment:
Note: Response not streaming is not an issue - we don't download anything big in the client.